app/shared/utils/Snackbar.js

```diff
@@ -22,6 +25,30 @@ const Alert = forwardRef(function Alert(props, ref) {
 export default function Snackbar() {
   const state = useHookstate(globalSnackbarState);
   const snackbarState = state.get();
+  const params = useSearchParams();
+  const pathname = usePathname();
+
+  useEffect(() => {
+    const redirMsg = params.get(REDIRECT_MSG_PARAM);
+
+    if (redirMsg) {
+      // remove redirect message param
+      const newParams = new URLSearchParams(params);
+      newParams.delete(REDIRECT_MSG_PARAM);
+      window.history.replaceState(
+        null,
+        undefined,
+        pathname + '?' + newParams.toString(),
+      );
+
+      // Show message after user has been redirected
+      state.set(() => ({
+        isOpen: true,
+        message: redirMsg,
+        isError: false,
+      }));
+    }
+  }, [params, pathname]);
```

Comment:
No, that is not the responsibility of snackbar.
you can add a callback that fires after something happens here, but snackbar should not know about redirect, window history etc.